tgui/packages/tgui/interfaces/CrewConsole.jsx

```diff
@@ -86,25 +87,119 @@ export const CrewConsole = () => {
 
 const CrewTable = (props) => {
   const { act, data } = useBackend();
-  const sensors = sortBy(data.sensors ?? [], (s) => s.ijob);
+  const sensors = data.sensors ?? [];
+
+  const [shownSensors, setShownSensors] = useState(sensors.slice());
+  const [sortColumn, setSortColumn] = useState('ijob');
+  const [sortAsc, setSortAsc] = useState(true);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const updateSortMode = (column) => {
+    if (column === sortColumn) {
+      if (!sortAsc) {
+        setSortColumn('ijob');
+        setSortAsc(true);
+      } else {
+        setSortAsc(false);
+      }
+    } else {
+      setSortColumn(column);
+      setSortAsc(true);
+    }
+  };
+
+  const healthSort = (a, b) => {
+    if (a.life_status < b.life_status) return -1;
+    if (a.life_status > b.life_status) return 1;
+    if (a.health > b.health) return -1;
+    if (a.health < b.health) return 1;
+    return 0;
+  };
+
+  useEffect(() => {
+    let unsorted = sensors.slice();
+    if (sortColumn === 'ijob') {
+      let sorted = sortBy(data.sensors ?? [], (s) => s.ijob).filter((crew) =>
+        crew.name.toLowerCase().includes(searchQuery.toLowerCase()),
+      );
+      setShownSensors(sorted);
+      return;
+    } else if (sortColumn === 'health') {
+      unsorted.sort(healthSort);
+    } else {
+      unsorted.sort((a, b) => {
+        a[sortColumn] ??= '~';
+        b[sortColumn] ??= '~';
+        if (a[sortColumn] < b[sortColumn]) return -1;
+        if (a[sortColumn] > b[sortColumn]) return 1;
+        return 0;
+      });
+    }
+
+    let sorted = unsorted.filter((crew) =>
+      crew.name.toLowerCase().includes(searchQuery.toLowerCase()),
+    );
```

Comment:
`createSearch`
should look like this
`unsorted.filter(createSearch(searchQuery, (crew) => crew.name);` (standard lib)
or
`filter(unsorted, createSearch(searchQuery, (crew) => crew.name));` (collections)